lib/node/nodes/intersection.js

```diff
@@ -1,28 +1,41 @@
 'use strict';
 
 var Node = require('../node');
+var debug = require('../../util/debug')('analysis:intersection');
 
 var TYPE = 'intersection';
 var PARAMS = {
     source: Node.PARAM.NODE(Node.GEOMETRY.ANY),
-    target: Node.PARAM.NODE(Node.GEOMETRY.ANY)
+    source_columns: Node.PARAM.NULLABLE(Node.PARAM.ARRAY(Node.PARAM.STRING()), []),
+    target: Node.PARAM.NODE(Node.GEOMETRY.ANY),
 };
 
-var Intersection = Node.create(TYPE, PARAMS, { cache: true, version: 3 });
+var Intersection = Node.create(TYPE, PARAMS, {
+    cache: true,
+    version: 4,
+    beforeCreate: function (node) {
```

Comment:
Not a high priority thing, just wanted to be sure it was not actually needed. Also wanted to let you know the intentional behaviour for `beforeCreate` 😉 .